lib/cryptoexchange/exchanges/binance_futures/services/contract_stat.rb

```diff
@@ -9,26 +9,30 @@ def supports_individual_ticker_query?
         end
 
         def fetch(market_pair)
-          index = super(index_url(market_pair))
+          contract_info = super(contract_info_url(market_pair))
 
-          adapt(nil, index, market_pair)
+          adapt(nil, contract_info, market_pair)
         end
 
         # def open_interest_url(market_pair)
         #   TODO: pending
         # end
 
-        def index_url(market_pair)
+        def contract_info_url(market_pair)
           "#{Cryptoexchange::Exchanges::BinanceFutures::Market::API_URL}/premiumIndex?symbol=#{market_pair.inst_id}"
         end
 
-        def adapt(open_interest, index, market_pair)
+        def adapt(open_interest, contract_info, market_pair)
           contract_stat = Cryptoexchange::Models::ContractStat.new
           contract_stat.base      = market_pair.base
           contract_stat.target    = market_pair.target
           contract_stat.market    = BinanceFutures::Market::NAME
-          contract_stat.index     = index['markPrice'].to_f
-          contract_stat.payload   = { "index" => index }
+          contract_stat.index     = contract_info['markPrice'].to_f
+          contract_stat.funding_rate_percentage     = contract_info['lastFundingRate'].to_f * 100
+          contract_stat.next_funding_rate_timestamp     = contract_info['nextFundingTime']/1000
+          contract_stat.funding_rate_percentage_predicted = nil
+          contract_stat.contract_type = "perpetual"
+          contract_stat.payload   = { "contract_info" => contract_info }
```

Comment:
curious why we want to wrap it in "contract_info"

Reply:
actually we don't need to wrap it in `contract_info` in this case. The reason we wrap it in `open_interest`, `index` or `contract_info` in other cases is because the payload might came from different api endpoint it is easier to identify them this way. Will amend this.